src/nodeTypesStore.js

```diff
@@ -51,10 +54,41 @@ export default new Vuex.Store({
         return node.label.toLowerCase().includes(searchTerm.toLowerCase());
       });
     },
+    setPmBlockNodeTypes(state, pmBlockNodeTypes) {
+      state.pmBlockNodeTypes = pmBlockNodeTypes
+        .filter(pmBlockNode => pmBlockNode.control)
+        .map(pmBlockNode => ({
+          id: uniqueId('pmBlockNode_'),
+          type: pmBlockNode.id,
+          icon: pmBlockNode.icon,
+          label: pmBlockNode.label,
+          description: pmBlockNode.description,
+          bpmnType: pmBlockNode.bpmnType,
+          rank: pmBlockNode.rank || BOTTOM,
+          items: pmBlockNode.items?.map(item => ({ ...item, type: item.id })),
+        }))
+        .sort((node1, node2) => node1.rank - node2.rank);
+    },
+    setFilteredPmBlockNodeTypes(state, searchTerm) {
+      // TODO: Configure Pm Block search bar
+      console.log('FILTER PM BLOCK NODE TYPES', state, searchTerm);
+      // const nodeTypes = state.nodeTypes;
+      // state.searchTerm = searchTerm;
+      // const allNodes = uniqBy([...pinnedNodeTypes, ...nodeTypes], 'id');
+      // state.filteredNodeTypes = allNodes.filter(node => {
+      //   return node.label.toLowerCase().includes(searchTerm.toLowerCase());
+      // });
+    },
     clearFilteredNodes(state) {
       state.filteredNodeTypes = [];
       state.searchTerm = '';
     },
+    clearFilteredPmBlockNodes(state) {
+      // TODO: Clear Pm Block search bar
+      console.log('CLEAR FILTER PM BLOCK NODE', state);
+      // state.filteredNodeTypes = [];
+      // state.searchTerm = '';
+    },
```

Comment:
I would add some similar logic or maybe reuse the component I created for the searchBar and passing like a SeachBarType to switch between regular nodeTypes and PmBlocks. That way we can do some v-if in rendering or logic.
Then on switching tabs, I would call here in this function `state.filteredNodeTypes = []` to clear the search bar to 0